src/util/DataGenerator.py

```diff
@@ -80,7 +83,27 @@ def filter_data(df, case, num_params):
         selected_categorical_column_names = df.columns[num_params+1:y+51]
         combined_column_names += selected_categorical_column_names.tolist()
     return combined_column_names
-def build_plot(data: pd.DataFrame, combined_column_names: list, target, case, name):
+
+
+def encode_import_data(df):
+    label_encoder = LabelEncoder()
+
+    # Apply label encoding to 'sex', 'race', and 'ethnicity' columns
+    #df['patient_id'] = label_encoder.fit_transform(df['patient_id'])
+    df['patient_id'] = df['patient_id'].str.extract('(\d+)').astype(int)
+    df['sex'] = label_encoder.fit_transform(df['sex'])
+    df['race'] = label_encoder.fit_transform(df['race'])
+    df['ethnicity'] = label_encoder.fit_transform(df['ethnicity'])
```

Comment:
Why did you apply LabelEncoder for "race" and "ethnicity" if you apply on-hot encoding then? I think you can use only one-hot